v2/internal/controllers/crd_web_test.go

```diff
@@ -0,0 +1,97 @@
+/*
+Copyright (c) Microsoft Corporation.
+Licensed under the MIT license.
+*/
+
+package controllers_test
+
+import (
+	"testing"
+
+	resources "github.com/Azure/azure-service-operator/v2/api/resources/v1beta20200601"
+	"github.com/Azure/azure-service-operator/v2/api/web/v1beta20220301"
+	"github.com/Azure/azure-service-operator/v2/internal/testcommon"
+	"github.com/Azure/go-autorest/autorest/to"
+	"github.com/onsi/gomega"
+)
+
+func Test_Web_ServerFarm_CRUD(t *testing.T) {
+	t.Parallel()
+
+	tc := globalTestContext.ForTest(t)
+
+	rg := tc.CreateTestResourceGroupAndWait()
+
+	serverfarm := newServerFarm(tc, rg)
+
+	tc.CreateResourceAndWait(serverfarm)
+
+	armId := *serverfarm.Status.Id
+	old := serverfarm.DeepCopy()
+	serverfarm.Spec.PerSiteScaling = to.BoolPtr(true)
+	tc.PatchResourceAndWait(old, serverfarm)
+	tc.Expect(serverfarm.Status.PerSiteScaling).To(gomega.Equal(to.BoolPtr(true)))
+
+	tc.DeleteResourcesAndWait(serverfarm)
+
+	exists, _, err := tc.AzureClient.HeadByID(
+		tc.Ctx,
+		armId,
+		string(v1beta20220301.APIVersion_Value))
+	tc.Expect(err).ToNot(gomega.HaveOccurred())
+	tc.Expect(exists).To(gomega.BeFalse())
+
+}
+
+func Test_Web_Site_CRUD(t *testing.T) {
+	t.Parallel()
+
+	tc := globalTestContext.ForTest(t)
+
+	rg := tc.CreateTestResourceGroupAndWait()
+
+	serverfarm := newServerFarm(tc, rg)
+
+	site := &v1beta20220301.Site{
+		ObjectMeta: tc.MakeObjectMeta("function"),
+		Spec: v1beta20220301.Sites_Spec{
```

Comment:
So, according to [the documentation](https://docs.microsoft.com/en-us/azure/azure-functions/functions-create-first-function-resource-manager?tabs=azure-cli), if we want to use this API to make a Function, we need to allow users to set a value in `siteConfig.AppSettings` that looks like `"[format('DefaultEndpointsProtocol=https;AccountName={0};EndpointSuffix={1};AccountKey={2}', variables('storageAccountName'), environment().suffixes.storage, listKeys(resourceId('Microsoft.Storage/storageAccounts', variables('storageAccountName')), '2021-08-01').keys[0].value)]`

The issue is that we don't have a way to do that securely, and we also don't have a way to let them dynamically build that string (since AFAIK we don't support exporting such a field on the storage side).

We could add the ability to export such a field on the storage side but that still leaves us with an inability to specify a connection string here in a secure reference-based way.

Maybe we need to think about somehow being able to tag the `KeyValuePair` type or whatever it's called in `AppSettings` to allow either a raw string _or_ a reference? That's not support we have today though.

Reply:
Holding this off until a user actually asks for it and have a user scenario. For now have added a TODO in here to track.